composer/core/state.py

```diff
@@ -343,57 +343,44 @@ def schedulers(self):
     def schedulers(self, schedulers: Union[types.PyTorchScheduler, Sequence[types.PyTorchScheduler]]):
         self._schedulers[:] = ensure_tuple(schedulers)
 
-    def batch_get_item(self, key: Optional[Any] = None, get_fn: Optional[Callable[[Any], Any]] = None) -> Any:
+    def batch_get_item(self, key: Union[Any, Callable, Sequence[Callable[[], Any]]]) -> Any:
```

Comment:
just speculating: should we employ some python magic to make `state.batch` support the get_item and set_item methods? so one could do `state.batch[key]` and `state.batch[key] = value`? Or would that be too magical?

Reply:
That is perhaps a bit too magic. It would be cool, but might not be worth implementing. Unfortunately, properties in python only allow getters, setters, and deleters, so we can't do `@batch.setitem` or `@batch.getitem`, even though if we made batch a property, we could do `@batch.setter`

 We would have to either:
a. wrap batch in a class, where we implement our own `__getitem__` and `__setitem__`
b. monkeypatch in our own `__getitem__` and `__setitem__`

Maybe in a future PR? 
@ravi-mosaicml, you have any ideas on how to do this?

Reply:
While wrapping would work, I'd prefer not to, since that would likely break things if a user did `state.model(state.batch)` without first unwrapping it. Regardless if we wrapped it or monkeypatched the item directly, most batch objects already have a `__getitem__` implementation, for which I think it'd feel weird to override the default.

(In most cases, anyways, the batch_get_item heleper just does `state.batch[key]`...)